File: Daily/sbgo/Leetcode690-员工的重要性.go
package sbgo

func getImportance(employees []*Employee, id int) int {
	res := 0
	var dfs func(*Employee)
	dfs = func(ep *Employee) {
		res += ep.Importance
		if len(ep.Subordinates) == 0 {
			return
		}
		for _, idx := range ep.Subordinates {
			for _, epl := range employees {
				if idx == epl.Id {
					dfs(epl)
				}
			}
		}
		return
	}
	for i := range employees {
		if employees[i].Id == id {
			dfs(employees[i])
		}
	}
	return res
}

File: Leetcode/PY/Leetcode690-员工的重要性.py
class Solution:
    def __init__(self):
        self.res = 0
        self.employees = None
    def getImportance(self, employees: List['Employee'], id: int) -> int:
        self.employees = employees
        self.res = 0
        def dfs(epl):
            self.res += epl.importance
            if not epl.subordinates:
                return 
            for ep in epl.subordinates:
                for worker in self.employees:
                    if worker.id == ep:
                        dfs(worker)
        for epl in employees:
            if epl.id == id:
                dfs(epl)
                break
        return self.res